intermediate_source/pinmem_nonblock.py

```diff
@@ -0,0 +1,429 @@
+# -*- coding: utf-8 -*-
+"""
+A guide on good usage of `non_blocking` and `pin_memory()` in PyTorch
+=====================================================================
+
+TL;DR
+-----
```

Comment:
I suggest we remove the TL;DR - the text below can be just the intro abstract right after the title. Can you follow this template: https://github.com/pytorch/tutorials/blob/main/beginner_source/template_tutorial.py  
Add What you will learn and Prerequisites and Author

Reply:
Sure!